### ISSUE TITLE
Typing interrupted by preview updates

### ISSUE THREAD

**Issue by ocdtrekkie** [jacksingleton/hacker-slides#21]
Okay, trying to type in a Hacker Slides, and this is killing me. Every time I type like... a letter, the next two letters aren't recognized because the program is busy re-rendering the preview with the last letter in it. I've taken to typing in Notepad and copy/pasting lines into the Hacker Slides.

**Comment by jacksingleton** [comment]
Do you know if the text gets saved, even if it doesn't show in the preview?

Try waiting a second, add one letter, and see if the preview catches up with all the text it missed.

If so I'll collapse this issue into https://github.com/jacksingleton/hacker-slides/issues/6.

It definitely needs to be fixed :) The way save/preview currently works is very hacky and naive. It falls down when any significant latency is present.

**Comment by ocdtrekkie** [comment]
When I start typing a word, while it's generating a preview, the missing letters never show up. I'll just get the letters after it finished generating the preview. Of course, then I have to wait for that preview to generate before I can delete those letters to fix it. It's a vicious cycle. ;)

**Comment by ocdtrekkie** [comment]
And the issue isn't with it being missed in the preview, it's missed in the input window entirely. I think it's a separate issue.

**Comment by larjona** [comment]
I'm having this issue too. My work around is creating my slides in my local editor (with markdown highlight) and select_all+copy+paste_in_hacker_slides from time to time.

**Comment by JonTheNiceGuy** [comment]
Likewise. Is there anything I can do to help here?

**Comment by jacksingleton** [comment]
I've changed slide rendering to happen only after a save completes successfully. Let me know if this improves things!

**Comment by erdii** [comment]
To me typing is still extremly laggy :(
Chars are dropped at 50% of keypresses when I type fast.

I recommend increasing the delay time for saving in [`static/js/save.js`:18](https://github.com/jacksingleton/hacker-slides/blob/master/static/js/save.js#L18) to 500 or even more ms or adding an option at runtime to toggle saving at all.

**Comment by erdii** [comment]
Actually the issue IS already fixed! I'm sorry!
It's just that the package from Sandstorm App Market is outdated :(

**Comment by jacksingleton** [comment]
@ocdtrekkie, @larjona, @JonTheNiceGuy 

Could you let me know if this is still an issue for you and if so what browser you are using?

And if you have time, it would be super helpful to know if this issue is fixed in the new "plain text mode" -- which I just pushed out a few days ago.

**Comment by JonTheNiceGuy** [comment]
Both states work like a dream on Firefox for Android. I'll try to test from Ubuntu with Firefox and Chrome tomorrow.

**Comment by erdii** [comment]
On OS X  the perf is good on Firefox (latest) and Chrome (latest).
Still have to test with Safari, IE and on Mobile

**Comment by larjona** [comment]
Works for me in Debian Stable + Iceweasel (Firefox) 38.5.

Thanks!

**Comment by jacksingleton** [comment]
Thanks for the feedback everyone! I'm going to close this now, feel free to open another issue if you run into problems.

**Comment by lotcz** [comment]
Hello, I am afraid this issue is back. In Firefox it is eating characters that are written during the refresh, in Chrome it is little better, but still very laggy.
